server/src/main/java/org/opensearch/cluster/ApplicationManager.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster;
+
+import java.security.Principal;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.identity.scopes.Scope;
+
+/**
+ * The ApplicationManager class handles the processing and resolution of multiple types of applications. Using the class, OpenSearch can
+ * continue to resolve requests even when specific application types are disabled. For example, the ExtensionManager can be Noop in which case
+ * the ApplicationManager is able to resolve requests for other application types still
+ *
+ * @opensearch.experimental
+ */
+public class ApplicationManager {
```

Comment:
The terminology is a bit confusing: we are introducing the scopes for extensions but are referring to those as extensions. I think this class could be eliminated in favour of  `ExtensionsManager` providing the  scopes (since the only thing this class does is to reroute to `ExtensionsManager`)

Reply:
Hi @reta, thank you for taking a look. The ApplicationManager allows us to connect Extensions and Plugins as we can see here: https://github.com/opensearch-project/OpenSearch/pull/8526. I separated the two PRs because I did not want to make things too complicated for review. I can leave it till the later PR if you prefer, but I do think we will want to have an ApplicationManager 


What do you prefer.

Reply:
>  I can leave it till the later PR if you prefer, but I do think we will want to have an ApplicationManager

I think the role of `ApplicationManager` is very unclear (in this change set), if we could get back to it  later on (next PR as you mentioned), that would be better,  thank you.

Reply:
I will remove it.